### PR TITLE
BUG: Fix for pd.read_fwf not reading addinfourl object

### LINKED ISSUE #26376: pd.read_fwf fails with file pointer to url
#### Code Sample, a copy-pastable example if possible

```python
# Your code here

import urllib.request
import pandas as pd

url='ftp://ftp.ncdc.noaa.gov/pub/data/igra/igra2-station-list.txt'
urllib.request.urlretrieve(url,url.split('/')[-1])
igrainv=pd.read_fwf(url.split('/')[-1],widths=(2,1,3,5,9,10,7,4,30,5,5,7),
                        names=('CC','Network','Code','StationId','Latitude','Longitude','Elev','dummy','StationName','From','To','Nrec'))

f=urllib.request.urlopen(url)
igrainv=pd.read_fwf(f,widths=(2,1,3,5,9,10,7,4,30,5,5,7),
                        names=('CC','Network','Code','StationId','Latitude','Longitude','Elev','dummy','StationName','From','To','Nrec'))

```
#### Problem description

The first read_fwf from a file downloaded from the internet works, the second read_fwf of the same file using an url pointer fails.
I get the following error:

Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 803, in read_fwf
    return _read(filepath_or_buffer, kwds)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 429, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 895, in __init__
    self._make_engine(self.engine)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 1132, in _make_engine
    self._engine = klass(self.f, **self.options)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 3605, in __init__
    PythonParser.__init__(self, f, **kwds)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 2238, in __init__
    self.unnamed_cols) = self._infer_columns()
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 2614, in _infer_columns
    line = self._buffered_line()
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 2689, in _buffered_line
    return self._next_line()
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 2794, in _next_line
    orig_line = self._next_iter_line(row_num=self.pos + 1)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 2854, in _next_iter_line
    return next(self.data)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py", line 3589, in __next__
    line = next(self.f)
TypeError: 'addinfourl' object is not an iterator

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.0.final.0
python-bits: 64
OS: Linux
OS-release: 2.6.32-754.10.1.el6.centos.plus.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: 4.4.2
pip: 19.1.1
setuptools: 41.0.1
Cython: 0.29.7
numpy: 1.16.3
scipy: 1.2.1
pyarrow: None
xarray: 0.12.1
IPython: 7.5.0
sphinx: 2.0.1
patsy: 0.5.1
dateutil: 2.8.0
pytz: 2019.1
blosc: None
bottleneck: 1.2.1
tables: 3.5.1
numexpr: 2.6.9
feather: None
matplotlib: 3.0.2
openpyxl: 2.6.2
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: 1.1.8
lxml.etree: 4.3.3
bs4: 4.7.1
html5lib: 1.0.1
sqlalchemy: 1.3.3
pymysql: None
psycopg2: 2.7.5 (dt dec pq3 ext lo64)
jinja2: 2.10.1
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by shantanu-gontia**
- [x] closes #26376 
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

----

Not sure, how to add a new test for this particular case. Is it is possible without creating a `addinfourl` object with a specific URL? Thanks.

### CODE REVIEW

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -417,6 +417,7 @@ I/O
 - Fixed memory leak in :meth:`DataFrame.to_json` when dealing with numeric data (:issue:`24889`)
 - Bug in :func:`read_json` where date strings with ``Z`` were not converted to a UTC timezone (:issue:`26168`)
 - Added ``cache_dates=True`` parameter to :meth:`read_csv`, which allows to cache unique dates when they are parsed (:issue:`25990`)
+- Fixed error in :meth:``pandas.read_fwf`` where passing a :class:``urllib.response.addinfourl`` throws an error. (:issue:`26376`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.25.0.rst]: I would replace the "class urllib...forurl" with "file pointer to URL" as in the issue title.

**Comment by gfyoung** [comment]
> Is it is possible without creating a addinfourl object with a specific URL? Thanks.

@leohaim : Can you confirm whether this patch addresses your issue?

@jreback @TomAugspurger : Don't we have remote data files we could hit for a test like this?

File: pandas/io/parsers.py [hunk 2]
```diff
@@ -3606,7 +3606,12 @@ def __next__(self):
                 self.buffer = None
                 line = next(self.f)
```
Comment (jreback) [inline pandas/io/parsers.py]: I think you should just be able to use ``.readline()`` always here (and above); file iterators have this property.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/parsers.py (hunk 2 above)
Comment (shantanu-gontia) [reply pandas/io/parsers.py]: updated in the next commit.

**Comment by jreback** [comment]
closing as stale, ping if you'd like to continue.

**Pull request closed without merging**